src/main/java/edu/umich/its/spe/SPEEsbImpl.java

```diff
@@ -0,0 +1,147 @@
+package edu.umich.its.spe;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import edu.umich.ctools.esb.utils.WAPI;
+import edu.umich.ctools.esb.utils.WAPIResultWrapper;
+
+/* 
+ * Implement esb calls to get / put grade information.
+ * The information required is passed in a hashmap.  Some values come from the
+ * properties files.
+ */
+
+// Spring: this makes the class discoverable for autowiring.
+@Component
+
+public class SPEEsbImpl implements SPEEsb {
+
```

Comment:
Lot of hard coded values in these file are repeated more then once and can be made constants